src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -81,9 +82,19 @@ public TracerSettings(IConfigurationSource source)
                                    // default value
                                    false;
 
-            MaxTracesSubmittedPerSecond = source?.GetInt32(ConfigurationKeys.MaxTracesSubmittedPerSecond) ??
-                                          // default value
-                                          100;
+            var maxTracesPerSecond = source?.GetInt32(ConfigurationKeys.MaxTracesSubmittedPerSecond);
+
+            if (maxTracesPerSecond != null)
+            {
+                // Ensure our flag for the rate limiter is enabled
+                RuleBasedSampler.OptInTracingWithoutLimits();
+            }
+            else
+            {
+                maxTracesPerSecond = 100; // default
+            }
+
+            MaxTracesSubmittedPerSecond = maxTracesPerSecond.Value;
```

Comment:
If I am looking at it right I think this is perfect, but just wanted to drop a message here to double check.
We are doing `RuleBasedSampler.OptInTracingWithoutLimits();` here (if the parameter that controls traces per second is set) and when a new a new rule is registered (if the user register at least one rule).

Some languages let users opt-in also using of a global configuration env variable, `DD_TRACE_SAMPLE_RATE=0.7` which means, not surprisingly...sample 70% of the traces, no matter what. In this case, we need to enforce the traces per second limit as well.

I could not find evidence of this in the possibility in this tracer - I looked [here](https://github.com/DataDog/dd-trace-dotnet/blob/master/src/Datadog.Trace/Configuration/ConfigurationKeys.cs) - , but just wanted to double check. If you haven't this configuration option then no action is required, otherwise I'd ask you to make sure the if that env is set we comply with max traces per second.

Reply:
We don't have such a configuration option no, currently that would be accomplished with a single rule with no filter on service or operation.